x-pack/legacy/plugins/monitoring/public/components/alerts/configuration/configuration.tsx

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ReactNode } from 'react';
+import { kfetch } from 'ui/kfetch';
+import { EuiSteps } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ActionResult } from '../../../../../actions/server/types';
+import { ALERT_ACTION_TYPE_EMAIL } from '../../../../common/constants';
+import { getMissingFieldErrors } from '../../../lib/form_validation';
+import { Step1 } from './step1';
+import { Step2 } from './step2';
+import { Step3 } from './step3';
+
+export interface AlertsConfigurationProps {
+  emailAddress: string;
+  onDone: Function;
+}
+
+export interface StepResult {
+  title: string;
+  children: ReactNode;
+  status: any;
+}
+
+export interface AlertsConfigurationForm {
+  email: string | null;
+}
+
+export const NEW_ACTION_ID = '__new__';
+
+export const AlertsConfiguration: React.FC<AlertsConfigurationProps> = (
+  props: AlertsConfigurationProps
+) => {
+  const { onDone } = props;
+
+  const [emailActions, setEmailActions] = React.useState<ActionResult[]>([]);
+  const [selectedEmailActionId, setSelectedEmailActionId] = React.useState('');
+  const [editAction, setEditAction] = React.useState<ActionResult | null>(null);
+  const [emailAddress, setEmailAddress] = React.useState(props.emailAddress);
+  const [formErrors, setFormErrors] = React.useState<AlertsConfigurationForm>({ email: null });
+  const [showFormErrors, setShowFormErrors] = React.useState(false);
+  const [isSaving, setIsSaving] = React.useState(false);
+  const [saveError, setSaveError] = React.useState('');
```

Comment:
You can also use an object for something like this so you don't have to deal with all the "sets" eg:
```javascript
const [formState = setFormState] = {...defaultFormValuesHere}; //can also cast it as a type
if (!formState.emailAddress) {
   //...some logic here
   setFormState({...formState, formErrors: 'No email address'});
}
```
I guess it's more of a personal preference, so w/e makes more sense to you

Reply:
Yea, I actually prefer the more explicit getters/setters. I'll leave it as-is unless you had a stronger reason to change it